versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/events/ReferenceEvent.java

```diff
@@ -21,12 +21,23 @@
 public interface ReferenceEvent extends AdapterEvent {
   NamedRef getRef();
 
-  Hash getHash();
+  /**
+   * Hash of the reference, specific meaning for each event type.
+   *
+   * <ul>
+   *   <li>For {@link ReferenceCreatedEvent}: the current/initial commit ID of the reference when it
+   *       was created
+   *   <li>For {@link ReferenceDeletedEvent}: the current commit ID of the reference when it was
+   *       deleted
+   *   <li>For {@link ReferenceAssignedEvent}): the <em>new</em> commit ID of the named reference
+   * </ul>
+   */
+  Hash getCurrentHash();
```

Comment:
+1 to method rename.

nit: WDYT about moving the class-specific javadoc to overrides in each of those classes?